core/controllers/reader_test.py

```diff
@@ -280,6 +280,45 @@ def test_get_exploration_pretests(self):
             expected_status_int=400)
 
 
+class QuestionsUnitTest(test_utils.GenericTestBase):
```

Comment:
Lets add a few failing tests as well:
- when skill_id isn't available
- when the acl decorator rejects the user
- when question count is 0
- Test that we return max questions available if question_count > max_questions_available

Reply:
added the tests